lib/sass/file-to-class-names.ts

```diff
@@ -5,88 +5,76 @@ import { sourceToClassNames } from "./source-to-class-names";
 import { Implementations, getImplementation } from "../implementations";
 import { customImporters, Aliases, SASSImporterOptions } from "./importer";
 
+export { Aliases };
 export type ClassName = string;
-export type ClassNames = ClassName[];
+interface Transformer {
+  (className: ClassName): string;
+}
+
+const transformersMap = {
+  camel: (className: ClassName) => camelCase(className),
+  dashes: (className: ClassName) =>
+    /-/.test(className) ? camelCase(className) : className,
+  kebab: (className: ClassName) => transformersMap.param(className),
+  none: (className: ClassName) => className,
+  param: (className: ClassName) => paramCase(className),
+  snake: (className: ClassName) => snakeCase(className),
+} as const;
 
-export type NameFormat = typeof NAME_FORMATS[number];
+export type NameFormatInput = keyof typeof transformersMap | "all" | "none";
+
+export const NAME_FORMATS = Object.keys(transformersMap).concat([
+  "all",
+]) as NameFormatInput[];
+
+type NameFormatsWithTransformer = Exclude<NameFormatInput, "all">;
```

Comment:
this feels like duplicative work (first add `all` above, then remove `all`). can we instead start with this type inferred from the object, then the super type can reuse it and add `all`? thinking something like the following

```tsx
type NameFormatsWithTransformer = keyof typeof transformersMap;
export type NameFormatInput = NameFormatsWithTransformer | "all";
```

Reply:
I'll have a look! 👍

Reply:
Addressed: [e67879f](https://github.com/skovy/typed-scss-modules/pull/164/commits/e67879fbfbadddc3b495be59a6183d29aa90b7bd)